src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -10003,19 +10004,45 @@ void StorageReplicatedMergeTree::backupData(
 
     /// Send a list of mutations to the coordination too (we need to find the mutations which are not finished for added part names).
     {
-        auto zookeeper = getZooKeeper();
+        const fs::path mutations_node_path = fs::path(zookeeper_path) / "mutations";
+        zkutil::ZooKeeperPtr zookeeper;
+
+        bool exists = false;
         Strings mutation_ids;
-        if (zookeeper->tryGetChildren(fs::path(zookeeper_path) / "mutations", mutation_ids) == Coordination::Error::ZOK)
+        {
+            ZooKeeperRetriesControl retries_ctl("getMutations", zookeeper_retries_info, nullptr);
+            retries_ctl.retryLoop([&]()
+            {
+                if (!zookeeper || zookeeper->expired())
```

Comment:
Why not do `zookeeper = local_context->getZooKeeper();` directly? It should handle all current and future use cases of session expiration.

Another option  I see used elsewhere is `renewZooKeeper`

Reply:
> Why not do zookeeper = local_context->getZooKeeper(); directly?

Here maybe, but below were we do tryGet in a loop, it will mean we'll need to go to context and lock it's mutex on each loop iteration, which is not a big problem, but why do that if it can be avoided, so here I made the same just for consistency, but it is changeable.

> Another option I see used elsewhere is renewZooKeeper

Yeah, I saw that, but it is a part of `WithRetries` class, which also requires `keeper_settings` and their creation is a bit too much it seems + I saw other places where `ZooKeeperRetriesControl ` is used like I use here, so I think it is ok.